### ISSUE TITLE
Building justbuild against fmt 9.1.x fails due to consteval issue

### ISSUE THREAD

**Issue by mhthies** [just-buildsystem/justbuild#10]
I'm trying to package *justbuild* for ArchLinux, linking against native shared libraries, provided by ArchLinux packages, as far as possible. The fmt library is currently provided in version 9.1.0 [in the ArchLinux package repositories](https://archlinux.org/packages/extra/x86_64/fmt/).

Building *justbuild* against this library version, using the `bootstrap.py` with `PACKAGE=YES` and Clang 15.0.7 fails with the following error message:

```txt
Running ['/usr/bin/clang++', '-std=c++20', '-DBOOTSTRAP_BUILD_TOOL', '-I', '/tmp/tmps83ihjbv/src', '-I', '/tmp/tmps83ihjbv/dep_includes', '-I', '/usr/include', '-c', '/tmp/tmps83ihjbv/src/src/utils/cpp/file_locking.cpp', '-o', '/tmp/tmps83ihjbv/src/src/utils/cpp/file_locking.o'] in '/tmp/tmps83ihjbv/src'
In file included from /tmp/tmps83ihjbv/src/src/utils/cpp/file_locking.cpp:19:
In file included from /tmp/tmps83ihjbv/src/src/buildtool/file_system/file_system_manager.hpp:32:
/tmp/tmps83ihjbv/src/src/buildtool/logging/logger.hpp:130:37: error: call to consteval function 'fmt::basic_format_string<char, std::basic_string<char>, std::basic_string<char>, const char *>::basic_format_string<std::basic_string<char>, 0>' is not a constant expression
            auto fmsg = fmt::format(msg, std::forward<T_Args>(args)...);
                                    ^
/tmp/tmps83ihjbv/src/src/buildtool/logging/logger.hpp:93:13: note: in instantiation of function template specialization 'Logger::FormatAndForward<std::basic_string<char>, std::basic_string<char>, const char *>' requested here
            FormatAndForward(nullptr,
            ^
/tmp/tmps83ihjbv/src/src/buildtool/file_system/file_system_manager.hpp:87:21: note: in instantiation of function template specialization 'Logger::Log<std::basic_string<char>, std::basic_string<char>, const char *>' requested here
            Logger::Log(LogLevel::Error,
                    ^
/tmp/tmps83ihjbv/src/src/buildtool/logging/logger.hpp:130:37: note: function parameter 'msg' with unknown value cannot be used in a constant expression
            auto fmsg = fmt::format(msg, std::forward<T_Args>(args)...);
                                    ^
/tmp/tmps83ihjbv/src/src/buildtool/logging/logger.hpp:120:53: note: declared here
                                 std::string const& msg,
                                                    ^
```

I was able to fix this error, based on this StackOverflow answer: https://stackoverflow.com/a/68675384/10315508 :

I replaced `fmt::format()` with `fmt::vformat()` and `std::forward<T_Args>()` with `fmt::make_format_args()` in `Logger::FormatAndForward()`, as shown in [this commit](https://github.com/mhthies/justbuild/commit/1b05e0350d86b97356f50b7910e480424b153e3c). I have not checked whether this is compatible with fmt 7.x and other compilers. Thus, I'll leave it here as a proposal but not create a PR.

**Comment by aehlig** [comment]
Thanks for your detailed report including the commit. I don't think, we should care about `fmt 7.x` any more, as the unstable/testing versions of the major distributions, where packaging happens, already have `fmt 9.1` and for ad-hoc installations, building from source following the bundled instructions is always an option.

So my suggestion is to update `fmt` and include the needed fixes. Are you happy with #15 (including the co-author attribution)?

**Comment by mhthies** [comment]
Thank you in return for taking care of my reports so quickly. :)

As written in my comment in the PR, this issue is fixed with the proposed changes. Of course, you can mention me as a co-author, if you consider this as appropriate.

**Comment by aehlig** [comment]
Fixed by 4d5724b6d71f5251f8945056f278b3142f66f1b2
